compiler/fir/checkers/src/org/jetbrains/kotlin/fir/analysis/checkers/extended/RedundantExplicitTypeChecker.kt

```diff
@@ -42,27 +43,27 @@ object RedundantExplicitTypeChecker : FirMemberDeclarationChecker() {
             is FirConstExpression<*> -> {
                 when (initializer.source?.elementType) {
                     KtNodeTypes.BOOLEAN_CONSTANT -> {
-                        if (type.classId != StandardClassIds.Boolean) return
+                        if (!type.isSame(StandardClassIds.Boolean)) return
```

Comment:
Please take into account (here and below) that redundant type should not be annoteted. Example:

```
@Target(AnnotationTarget.TYPE)
annotation class A

fun foo() {
    val x: @A Int /* NOT redundant */ = 1
}

```